pygfx/renderers/wgpu/_shaderlib.py

```diff
@@ -24,16 +24,13 @@ def light_deps_basic(self):
         return """
         // TODO smoothstep and saturate probably exists when Naga gets updated
         fn _smoothstep( low : f32, high : f32, x : f32 ) -> f32 {
-            let t = clamp( ( x - low ) / ( high - low ), 0.0, 1.0 );
+            let t = saturate( ( x - low ) / ( high - low ));
             return t * t * ( 3.0 - 2.0 * t );
         }
-        fn _saturate( x: f32) -> f32 {
-            return clamp(x, 0.0, 1.0);
-        }
         fn getDistanceAttenuation(light_distance: f32, cutoff_distance: f32, decay_exponent: f32) -> f32 {
             var distance_falloff: f32 = 1.0 / max( pow( light_distance, decay_exponent ), 0.01 );
             if ( cutoff_distance > 0.0 ) {
-                distance_falloff *= pow2( _saturate( 1.0 - pow4( light_distance / cutoff_distance ) ) );
+                distance_falloff *= pow2( saturate( 1.0 - pow4( light_distance / cutoff_distance ) ) );
```

Comment:
Then we can remove the `_saturate()` function defined above as well.

Reply:
@panxinmiao I was wondering whether you deliberately avoided the wgsl builtin `saturate` here.

Reply:
> Then we can remove the `_saturate()` function defined above as well.

I did that correct?